core/src/main/scala/org/apache/spark/internal/config/History.scala

```diff
@@ -79,6 +79,21 @@ private[spark] object History {
     .stringConf
     .createOptional
 
+  object LocalStoreSerializer extends Enumeration {
+    val JSON, PROTOBUF = Value
+  }
+
+  val LOCAL_STORE_SERIALIZER = ConfigBuilder("spark.history.store.serializer")
+    .doc("Serializer for writing/reading in-memory UI objects to/from disk-based KV Store; " +
+      "JSON or PROTOBUF. JSON serializer is the only choice before Spark 3.4.0, thus it is the " +
+      "default value. PROTOBUF serializer is fast and compact, and it is the default " +
+      "serializer for disk-based KV store of live UI.")
```

Comment:
I'm a bit confused about this config. SHS only reads the rocksdb files, but not write them. So it should support whatever format of the rocksdb values. Is there a way to encode the format in rocksdb as well?

Reply:
SHS writes RocksDB files during replaying event logs. The default serializer is JSON+GZIP, which is slower than the Protobuf serializer. Imagine that there are 300GB of event logs to replay...

Reply:
oh I see